### DISCUSSION TITLE
Parallel tests only running when enable_multithreading = false

### DISCUSSION THREAD

**Discussion by daleront** [getgauge/gauge#1899] [General]
I am using the dotnet version of gauge and have been running it for the last couple of years.

I am upgrading from gauge v1.0.6-> v1.1.6 and have come across a problem when running gauge.

When running the following command, when only updating libraries I am seeing that different Specs are writing over each other and trying to access each others thread.
`gauge run --parallel -n=4 --tags <tags> --env <env> --verbose <Spec folder>`

Then when I define `enable_multithreading = false` in the env.properties file, it is running as it does before with the correct number of threads that are not impeding on each other.

I followed the docs and it specifies that if you want to run parallel specs, then you have to have `enable_multithreading = true`. Is this correct anymore?

**Comment by zabil** [comment]
As mentioned at https://docs.gauge.org/execution.html#parallel-execution-by-using-threads

Gauge dotnet added multi threading support recently. 

Multi threading is a way to optimize parallel execution. Gauge can run tests in parallel without multi-threading. However, enabling multi-threading the code needs to be thread safe. This might explain why there were issues.
